crates/utils/src/rate_limit/mod.rs

```diff
@@ -133,20 +134,20 @@ where
     self.service.poll_ready(cx)
   }
 
-  fn call(&self, mut req: ServiceRequest) -> Self::Future {
+  fn call(&self, req: ServiceRequest) -> Self::Future {
     let ip_addr = get_ip(&req.connection_info());
 
-    let http_req = req.parts_mut().0.clone();
-    let fut = self
-      .rate_limited
-      .clone()
-      .wrap(ip_addr, self.service.call(req));
+    let rate_limited = self.rate_limited.clone();
+    let service = self.service.clone();
 
     Box::pin(async move {
-      let res = fut.await.map_err(actix_web::Error::from);
-      if let Some(r) = res? {
-        Ok(r)
+      let opt = rate_limited
+        .wrap(ip_addr, async move { Ok(()) as Result<(), LemmyError> })
```

Comment:
.wrap seems silly now that I pass it a fake future, though. it could just be `.check(ip_addr)` or something